sw/device/lib/dif/dif_i2c.h

```diff
@@ -0,0 +1,390 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_I2C_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_I2C_H_
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+
+/**
+ * Represents a speed setting for an I2C component: standard, fast, and
+ * fast plus, corresponding to 100 kbaud, 400 kbaud, and 1 Mbaud,
+ * respectively.
+ */
+typedef enum dif_i2c_speed {
```

Comment:
I explicitly asked he not include the representation in the enum, primarily because this
could lead to people thinking that they could pass arbitrary values where these are
used.

Reply:
Good point!